sdk/zksend/src/links.ts

```diff
@@ -69,41 +85,88 @@ export class ZkSendLinkBuilder {
 		return link.toString();
 	}
 
-	async addGasForClaim(
-		getAmount?: (options: {
-			mist: bigint;
-			objects: TransactionObjectInput[];
-			estimatedFee: bigint;
-		}) => Promise<bigint> | bigint,
-	) {
-		const estimatedFee = await this.#estimateClaimGasFee();
-		this.#gasFee = getAmount
-			? await getAmount({
+	async create({
+		signer,
+		...options
+	}: CreateZkSendLinkOptions & {
+		signer: Signer;
+	}) {
+		const txb = await this.createSendTransaction(options);
+
+		return this.#client.signAndExecuteTransactionBlock({
+			transactionBlock: await txb.build({ client: this.#client }),
+			signer,
+		});
+	}
+	async createSendTransaction({ calculateGas }: CreateZkSendLinkOptions = {}) {
+		const gasEstimateFromDryRun = await this.#estimateClaimGasFee();
+		const baseGasAmount = calculateGas
+			? await calculateGas({
 					mist: this.#mist,
 					objects: [...this.#objects],
-					estimatedFee,
+					gasEstimateFromDryRun,
 			  })
-			: estimatedFee;
-	}
+			: gasEstimateFromDryRun * 2n;
```

Comment:
Just gut checking that 2x seem like a sensible amount?

Reply:
I don't have a good idea of how much it fluctuates, but probably fine for now.  I don't think we need to go higher.  Lets leave it as is until we get some data that says we need to change it